include/h2o/url.h

```diff
@@ -121,4 +123,19 @@ inline h2o_iovec_t h2o_url_stringify(h2o_mem_pool_t *pool, const h2o_url_t *url)
     return h2o_url_resolve(pool, url, NULL, &tmp);
 }
 
+/**
+ * Compares to hostnames, taking into account whether they contain a
+ * unix path (the comparison will be case insensitive) or not.
+ */
+static inline int h2o_url_hosts_are_equal(const h2o_url_t *url_a, const h2o_url_t *url_b)
+{
+    if (url_a->host_is_unix_path != url_b->host_is_unix_path)
+        return 0;
+
+    if (!url_a->host_is_unix_path)
+        return h2o_lcstris(url_a->host.base, url_a->host.len, url_b->host.base, url_b->host.len);
+    else
+        return !strncasecmp(url_a->host.base, url_b->host.base, url_b->host.len);
```

Comment:
I assume we should use `h2o_memis` here, since file systems are generally case-sensitive, and to avoid out-of-bounds access if the length of the two strings are not equal (consider the case where `url_a->host.len < url_b->host.len`).

Note also that we generally refrain from using locale-sensitive functions like `strcasecmp`.